File: output/check_fncpat.py
# -*- coding: utf-8 -*-
import os, requests, json, urllib3
from requests.auth import HTTPBasicAuth
from dotenv import load_dotenv
from pathlib import Path
urllib3.disable_warnings()
load_dotenv(Path(__file__).parent.parent / ".env")
URL  = os.getenv("PRIORITY_URL_REAL", "").rstrip("/")
AUTH = HTTPBasicAuth(os.getenv("PRIORITY_USERNAME",""), os.getenv("PRIORITY_PASSWORD",""))
HDR  = {"Accept": "application/json", "OData-Version": "4.0"}

# 1. List FNCPAT patterns to find the reconciliation one
print("=== FNCPAT patterns ===")
r = requests.get(f"{URL}/FNCPAT?$top=20", headers=HDR, auth=AUTH, timeout=20, verify=False)
print(f"Status: {r.status_code}")
if r.ok:
    for p in r.json().get("value", []):
        print(f"  FNCPATNAME={p.get('FNCPATNAME')!r}  FNCPATDES={p.get('FNCPATDES','')!r}  FNCPATDES2={p.get('FNCPATDES2','')!r}")
else:
    print(r.text[:300])

# 2. Compare a recent "התאמה" entry vs simple entry
print("\n=== Recent FNCTRANS with FNCPATNAME ===")
r2 = requests.get(
    f"{URL}/FNCTRANS?$filter=FINAL eq 'Y'&$top=5&$select=FNCNUM,FNCPATNAME,FNCPATDES2,DETAILS,BRANCHNAME",
    headers=HDR, auth=AUTH, timeout=20, verify=False)
if r2.ok:
    for e in r2.json().get("value", []):
        print(f"  FNCNUM={e.get('FNCNUM')!r} PAT={e.get('FNCPATNAME')!r} PATDES={e.get('FNCPATDES2','')!r} DETAILS={str(e.get('DETAILS',''))[:30]!r}")
else:
    print(r2.text[:300])

File: backend/close_receipt/check_fncpat.js
'use strict';
const path = require('path');
require('dotenv').config({ path: path.join(__dirname, '../../.env') });

const odataUrl = (process.env.PRIORITY_URL_REAL || '').replace(/\/$/, '');
const user = process.env.PRIORITY_USERNAME;
const pass = process.env.PRIORITY_PASSWORD;
const auth = 'Basic ' + Buffer.from(`${user}:${pass}`).toString('base64');
const hdr  = { Authorization: auth, Accept: 'application/json', 'OData-Version': '4.0' };

async function get(url) {
  const r = await fetch(url, { headers: hdr });
  return r.json();
}

(async () => {
  // 1. List FNCPAT patterns
  console.log('=== FNCPAT patterns ===');
  try {
    const d = await get(`${odataUrl}/FNCPAT?$top=20`);
    (d.value || []).forEach(p => console.log(`  FNCPATNAME=${JSON.stringify(p.FNCPATNAME)}  DES=${JSON.stringify(p.FNCPATDES)}  DES2=${JSON.stringify(p.FNCPATDES2)}`));
  } catch(e) { console.log('  Error:', e.message); }

  // 2. Recent final FNCTRANS with FNCPATNAME
  console.log('\n=== Recent FINAL FNCTRANS with FNCPATNAME ===');
  try {
    const d2 = await get(`${odataUrl}/FNCTRANS?$filter=FINAL eq 'Y'&$top=5&$select=FNCNUM,FNCPATNAME,FNCPATDES2,DETAILS,BRANCHNAME`);
    (d2.value || []).forEach(e => console.log(`  ${e.FNCNUM} PAT=${JSON.stringify(e.FNCPATNAME)} PATDES=${JSON.stringify(e.FNCPATDES2)} DETAILS=${JSON.stringify((e.DETAILS||'').substring(0,30))}`));
  } catch(e) { console.log('  Error:', e.message); }
})();
